### PR TITLE
fix max app size error message

### PR DESCRIPTION
**Pull request by badescuga**
Due to the library that we are using (prettysize), we are not displaying the error message correctly when an app build exceeds app size limits for a specific build target.

### CODE REVIEW

File: src/util/byteStringFormatter.test.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import byteStringFormatter from './byteStringFormatter';
```
[the remark this answers is not in the document]
Comment (badescuga) [reply src/util/byteStringFormatter.test.ts]: hmm, sounds better, yes.. but `data` sounds really vague. `humanizeByteCount` / `humanizeByteSize`?

File: src/util/byteStringFormatter.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+export default function byteStringFormatter(
+  byteCount: number,
+  decimalCount = 2,
+) {
+  if (byteCount === 0) {
+    return '0 Bytes';
+  }
+
+  const unitSize = 1024;
+  const unitNames = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
```
[the remark this answers is not in the document]
Comment (badescuga) [reply src/util/byteStringFormatter.ts]: ok, going with `B`, `MB`, etc.

File: src/util/byteStringFormatter.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+function clamp(num: number, min: number, max: number) {
+  return num <= min ? min : num >= max ? max : num;
+}
+export default function byteStringFormatter(
+  byteCount: number,
+  decimalCount = 2,
+) {
+  if (byteCount === 0) {
+    return '0 Bytes';
```
[the remark this answers is not in the document]
Comment (badescuga) [reply src/util/byteStringFormatter.ts]: going with `B`

File: src/validateFileSizes.ts [hunk 4]
```diff
@@ -42,9 +42,10 @@ export default function validateFileSizes({
           category: DiagnosticCategory.Error,
           messageText: `${
             file.relative
-          } is larger than maximum allowed size. File size was ${prettysize(
```
[the remark this answers is not in the document]
Comment (badescuga) [reply src/validateFileSizes.ts]: I'd like to keep `4`. Want to limit more the possibility of a rounded value presenting misleading info.

File: src/util/byteStringFormatter.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (Hexxeh) [reply src/util/byteStringFormatter.test.ts]: `humanizeByteCount` sounds good!

File: src/validateFileSizes.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (Hexxeh) [reply src/validateFileSizes.ts]: Make it the default then? Doesn't look like the default is used anywhere right now, it's always overridden to 4.

File: src/util/humanizeByteCount.test.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+  it.each([[1048576, 0], [1048576, 2], [1048576, undefined], [1048576, 5]])(
+    'converts bytes to megabytes and shows no decimal point, even when we request it',
+    (byteCount, decimalCount) => {
+      expect(humanizeByteCount(byteCount, decimalCount)).toBe('1 MB');
+    },
+  );
+
+  it('handles decimal point accordingly when the conversion value is not an exact number', () => {
+    expect(humanizeByteCount(340000, 0)).toBe('332 KB');
```
Comment (Hexxeh) [inline src/util/humanizeByteCount.test.ts]: You can jest.each these too, just add a third parameter for the expected string.

**Comment by Hexxeh** [comment]
You'll need to sign-off this change using your Fitbit.com email address. Assuming your Git name/email config is correct, you can just do `git commit --signoff` using `--amend` as needed. That'll pass the DCO check.

**Comment by Hexxeh** [comment]
Looks like there's an issue on Node 8 specifically too.

**Pull request merged**